examples/imu/calibrate.cc

```diff
@@ -0,0 +1,70 @@
+/****************************************************************************
+ *                                                                          *
+ *  Copyright (C) 2020 RoboMaster.                                          *
+ *  Illini RoboMaster @ University of Illinois at Urbana-Champaign          *
+ *                                                                          *
+ *  This program is free software: you can redistribute it and/or modify    *
+ *  it under the terms of the GNU General Public License as published by    *
+ *  the Free Software Foundation, either version 3 of the License, or       *
+ *  (at your option) any later version.                                     *
+ *                                                                          *
+ *  This program is distributed in the hope that it will be useful,         *
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of          *
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the           *
+ *  GNU General Public License for more details.                            *
+ *                                                                          *
+ *  You should have received a copy of the GNU General Public License       *
+ *  along with this program. If not, see <http://www.gnu.org/licenses/>.    *
+ *                                                                          *
+ ****************************************************************************/
+
+#include <cstring>
+
+#include "bsp_gpio.h"
+#include "bsp_imu.h"
+#include "bsp_os.h"
+#include "bsp_uart.h"
+#include "cmsis_os.h"
+#include "main.h"
+
+#define ONBOARD_IMU_SPI hspi5
+#define ONBOARD_IMU_CS_GROUP GPIOF
+#define ONBOARD_IMU_CS_PIN GPIO_PIN_6
+#define PRING_UART huart8
+
+typedef struct {
+  char header;
+  bsp::vec3f_t acce;
+  bsp::vec3f_t gyro;
+  bsp::vec3f_t mag;
+  char terminator;
+} __attribute__((packed)) imu_data_t;
+
+static bsp::MPU6500* imu;
+static bsp::UART* uart;
+
+static imu_data_t imu_data;
+
+void RM_RTOS_Default_Task(const void* arguments) {
+  UNUSED(arguments);
+
+  bsp::GPIO chip_select(ONBOARD_IMU_CS_GROUP, ONBOARD_IMU_CS_PIN);
+  imu = new bsp::MPU6500(&ONBOARD_IMU_SPI, chip_select, MPU6500_IT_Pin);
+  uart = new bsp::UART(&huart8);
+  uart->SetupTx(sizeof(imu_data_t));
```

Comment:
The initialization could be moved into function `RM_RTOS_Init`

Reply:
After some thoughts, I feel like it is better to do allocation in the beginning of each tasks. This is because the allocations are made through FreeRTOS routines, and `RM_RTOS_Init` is a function that is executed outside of FreeRTOS scheduler. The allocated memory will not be available until the scheduler starts. It is safer to put it here as `RM_RTOS_Default_Task` is inside a FreeRTOS thread.